src/dependencies.js

```diff
@@ -92,25 +92,64 @@ const getFileDependencies = async function(path, packageJson, state) {
 // `require()` statements can be either `require('moduleName')` or
 // `require(path)`
 const getImportDependencies = function(dependency, basedir, packageJson, state) {
-  if (LOCAL_IMPORT_REGEXP.test(dependency)) {
-    return getLocalImportDependencies(dependency, basedir, packageJson, state)
+  if (LOCAL_IMPORT_REGEXP.test(dependency) || isTreeShakable(dependency)) {
+    return getTreeShakedDependencies(dependency, basedir, packageJson, state)
   }
 
-  return getModuleDependencies(dependency, basedir, state, packageJson)
+  return getAllDependencies(dependency, basedir, state, packageJson)
 }
 
 const LOCAL_IMPORT_REGEXP = /^(\.|\/)/
 
+const isTreeShakable = function(dependency) {
+  if (dependency.includes('next') && !TREE_SHAKABLE_DEPENDENCIES.has(dependency)) {
+    console.log(dependency)
+  }
+  return TREE_SHAKABLE_DEPENDENCIES.has(dependency)
+}
+
+const TREE_SHAKABLE_DEPENDENCIES = new Set([
```

Comment:
Based on bundling `Next.js` v10 and v9:
https://github.com/erezrokah/erez-test-next-bundling
https://github.com/erezrokah/erez-test-next9-bundling

Reply:
This seems very fragile as any new `next` require not on this list will result in bundling the entire framework.
However, for Next.js 9 this will work well since it is not updated frequently (I'm assuming only for security patches)

Reply:
Would the following idea help with the brittleness? 

Only keep in that list the files that directly required, not indirectly.
When a Function is directly requiring one of those files, it would be tree-shaked. However, any of the subsequent indirect `require()` within the same Node module would also be tree-shaked. 

This would require keeping some state indicating "currently tree-shaking Node module X" so the logic knows to keep tree-shaking any files from that Node module.

I am not sure whether this makes sense in our case. Would this help reduce the size of this list?
What do you think?

Reply:
That's a good suggestion. I did another test and everything on that list that's under `next/dist/next-server` is required directly (`next/dist/next-server/lib/i18n/normalize-locale-path` only when using `Next.js` 10).

Let me try you suggestion and see how it goes

Reply:
@ehmicky let me know if https://github.com/netlify/zip-it-and-ship-it/pull/242/commits/9716a604b1472e6e03b60cab05123fe425e91d2d is what you had in mind.
This works well for `Next.js` v9 and v10

Reply:
Thanks!
Are all of the following entry points used in the Next.js plugin?

Reply:
Those are all direct requires in `nextPage.js` (v9 or v10 and page or api)

Reply:
Discussed on Slack: what about this:
If we detect that a Function was bundled by `next-on-netlify`, then tree-shake _any_ `next/*` module.
We would need to have some way to detect that a Function was bundled by `next-on-netlify`.
But this would allow us not to rely on specific file paths.